plugin/src/main/java/io/jenkins/plugins/redis/RedisFingerprintStorage.java

```diff
@@ -0,0 +1,168 @@
+    }
+
+    public RedisFingerprintStorage() throws IOException {
+        try {
+            instanceId = Util.getDigestOf(new ByteArrayInputStream(InstanceIdentity.get().getPublic().getEncoded()));
+            createJedisPoolFromConfig();
+        } catch (IOException e) {
+            logger.log(Level.WARNING, "Failed to obtain Instance ID. " + e);
+            throw e;
+        }
+    }
+
+    void createJedisPoolFromConfig() {
+        GlobalRedisConfiguration config = GlobalRedisConfiguration.get();
+        createJedisPool(config.getHost(), config.getPort(), config.getConnectionTimeout(), config.getSocketTimeout(),
+                config.getUsername(), config.getPassword(), config.getDatabase(), config.getSsl());
+    }
+
+    void createJedisPool(
+            String host, int port, int connectionTimeout, int socketTimeout, String username, String password,
+            int database, boolean ssl) {
+        jedisPool = new JedisPool(new JedisPoolConfig(), host, port, connectionTimeout, socketTimeout, username,
+                password, database, "Jenkins", ssl);
+    }
+
+    private @NonNull Jedis getJedis() throws JedisException{
+        return jedisPool.getResource();
+    }
+
+    /**
+     * Saves the given fingerprint.
+     */
+    public synchronized void save(Fingerprint fp) throws JedisException {
+        Jedis jedis = null;
+        StringWriter writer = new StringWriter();
+        Fingerprint.getXStream().toXML(fp, writer);
+        try {
+            jedis = getJedis();
+            jedis.set(instanceId + fp.getHashString(), writer.toString());
+        } catch (JedisException e) {
+            logger.log(Level.WARNING, "Jedis failed in saving fingerprint: " + fp.toString() + ". " + e);
+            throw e;
+        } finally {
+            if (jedis != null) {
+                jedis.close();
+            }
+        }
+    }
+
+    /**
+     * Returns the fingerprint associated with the given unique id and the Jenkins instance ID, from the storage.
+     */
+    public @CheckForNull Fingerprint load(@NonNull String id) throws IOException, JedisException {
+        String loadedData;
+        Jedis jedis = null;
+
+        try {
+            jedis = getJedis();
+            loadedData = jedis.get(instanceId + id);
+        } catch (JedisException e) {
+            logger.log(Level.WARNING, "Jedis failed in loading fingerprint: " + id + e);
+            throw e;
+        } finally {
+            if (jedis != null) {
+                jedis.close();
+            }
+        }
+
+        if (loadedData == null) return null;
+
+        Object loadedObject = null;
+        Fingerprint loadedFingerprint;
+
+        try (InputStream in = new ByteArrayInputStream(loadedData.getBytes(StandardCharsets.UTF_8))) {
+            loadedObject = Fingerprint.getXStream().fromXML(in);
+            loadedFingerprint = (Fingerprint) loadedObject;
+        } catch (RuntimeException | Error e) {
```

Comment:
Errors are usually considered as fatal issues. Unless there is a known recoverable scenario, I would suggest to not catch them